test/tmisc.c

```diff
@@ -6264,14 +6287,230 @@ test_misc37(void)
+    /* Close file access property list */
+    ret = H5Pclose(fapl);
+    CHECK(ret, FAIL, "H5Pclose");
+
+    /* Close file */
+    ret = H5Fclose(fid);
+    CHECK(ret, FAIL, "H5Fclose");
+
+    /* Create objects with unusual datatypes and verify correct behavior */
+    for (u = 0; u < 3; u++) {
+        /* Create a file access property list */
+        fapl = H5Pcreate(H5P_FILE_ACCESS);
+        CHECK(fapl, FAIL, "H5Pcreate");
+
+        if (1 == u) {
+            /* Set property to allow unusual datatypes to be opened */
+            ret = H5Pset_relax_file_integrity_checks(fapl, H5F_RFIC_UNUSUAL_NUM_UNUSED_NUMERIC_BITS);
+            CHECK(ret, FAIL, "H5Pset_relax_file_integrity_checks");
+        }
+        else if (2 == u) {
+            /* Use a later version of the file format, with checksummed object headers */
+            ret = H5Pset_libver_bounds(fapl, H5F_LIBVER_LATEST, H5F_LIBVER_LATEST);
+            CHECK(ret, FAIL, "H5Pset_libver_bounds");
+        }
+
+        fid = H5Fcreate(MISC38C_FILE, H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
+        CHECK(fid, FAIL, "H5Fcreate");
+
+        /* Close file access property list */
+        ret = H5Pclose(fapl);
+        CHECK(ret, FAIL, "H5Pclose");
+
+        sid = H5Screate(H5S_SCALAR);
+        CHECK(sid, FAIL, "H5Screate");
+
+        tid = H5Tcopy(H5T_NATIVE_INT);
+        CHECK(tid, FAIL, "H5Tcopy");
+
+        /* Set type to have unusual size, for precision */
+        ret = H5Tset_size(tid, 1000);
+        CHECK(ret, FAIL, "H5Tset_size");
+
+        /* Create a dataset with the unusual datatype */
+        H5E_BEGIN_TRY
+        {
+            did = H5Dcreate2(fid, MISC38C_DSETNAME, tid, sid, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+        }
+        H5E_END_TRY
+        if (u > 0) {
+            CHECK(did, FAIL, "H5Dcreate2");
+
+            ret = H5Dclose(did);
+            CHECK(ret, FAIL, "H5Dclose");
+        }
+        else {
+            VERIFY(did, FAIL, "H5Dcreate2");
+        }
+
+        gid = H5Gopen2(fid, "/", H5P_DEFAULT);
+        CHECK(gid, FAIL, "H5Gopen2");
+
+        /* Create an attribute with the unusual datatype */
+        H5E_BEGIN_TRY
+        {
+            aid = H5Acreate2(gid, MISC38C_ATTRNAME, tid, sid, H5P_DEFAULT, H5P_DEFAULT);
+        }
+        H5E_END_TRY
+        if (u > 0) {
+            CHECK(aid, FAIL, "H5Acreate2");
+
+            ret = H5Aclose(aid);
+            CHECK(ret, FAIL, "H5Aclose");
+        }
+        else {
+            VERIFY(aid, FAIL, "H5Acreate2");
+        }
+
+        ret = H5Gclose(gid);
+        CHECK(ret, FAIL, "H5Gclose");
+
+        ret = H5Sclose(sid);
+        CHECK(ret, FAIL, "H5Sclose");
+
+        /* Create a committed datatype with the unusual datatype */
+        H5E_BEGIN_TRY
+        {
+            ret = H5Tcommit2(fid, MISC38C_TYPENAME, tid, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+        }
+        H5E_END_TRY
+        if (u > 0) {
+            CHECK(ret, FAIL, "H5Tcommit2");
+
+            ret = H5Tclose(tid);
```

Comment:
Should move this outside the if - else block so we close `tid` even if `H5Tcommit2` fails above

Reply:
Done